env0/resource_project.go

```diff
@@ -126,6 +136,29 @@ func resourceProjectDelete(ctx context.Context, d *schema.ResourceData, meta int
 
 	id := d.Id()
 
+	if d.Get("wait").(bool) {
+		ticker := time.NewTicker(PROJECT_DESTROY_WAIT_INTERVAL) // When invoked check if project can be deleted.
+		timer := time.NewTimer(PROJECT_DESTROY_TOTAL_WAIT_TIME) // When invoked wait time has elapsed.
+		done := make(chan bool)
+
+		go func() {
+			for {
+				select {
+				case <-timer.C:
+					done <- true
+					return
+				case <-ticker.C:
+					if resourceProjectAssertCanDelete(ctx, d, meta) == nil {
+						done <- true
+						return
+					}
+				}
+			}
+		}()
+
+		<-done
+	}
+
 	if err := resourceProjectAssertCanDelete(ctx, d, meta); err != nil {
```

Comment:
once done waiting (time elapsed or all environments deleted): check again if can delete.
If can't it will return an error (same as in the past).